src/Scripting/CSharp/Hosting/ObjectFormatter/CSharpTypeNameFormatter.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.Scripting.Hosting;
+
+namespace Microsoft.CodeAnalysis.CSharp.Scripting.Hosting
+{
+    public class CSharpTypeNameFormatter : CommonTypeNameFormatter
+    {
+        protected override CommonPrimitiveFormatter PrimitiveFormatter { get; }
```

Comment:
Is this intended to be part of hte public API?  Or is it just intended for us to override?  If the latter, 'internal' would be more appropriate.